packages/core/primitives/event-dispatch/src/register_events.ts

```diff
@@ -37,3 +38,32 @@ export function bootstrapEventContract(
   }
   return eventContract;
 }
+
+/**
+ * Provides a factory function for bootstrapping an event contract on a
+ * window object.
+ * @param field The property on the window that the event contract will be placed on.
+ * @param container The container that listens to events
+ * @param appId A given identifier for an application. If there are multiple apps on the page
+ *              then this is how contracts can be initialized for each one.
+ * @param eventTypes An array of event names that should be listened to.
+ * @param captureEventTypes An array of event names that should be listened to with capture.
+ * @param anyWindow The global window object that should receive the event contract.
+ * @returns The `event` contract. This is both assigned to `anyWindow` and returned for testing.
+ */
+export function bootstrapEarlyEventContract(
+  field: string,
+  container: HTMLElement,
+  appId: string,
+  eventTypes: string[],
+  captureEventTypes: string[],
+  earlyJsactionTracker: any = window,
```

Comment:
Can we use more specific type here?
```suggestion
  earlyJsactionTracker: EarlyJsactionDataContainer = window,
```

Reply:
the EarlyJsactionDataContainer is actually a field [field][appId] on the container.

Reply:
Would it be something like this? We can also move that into a separate `type` and add some docs. Do we have other places where such type might be needed?
```suggestion
  earlyJsactionTracker: {[key: string]: {[appId: string]: EarlyJsactionDataContainer} } = window,
```